### PR TITLE
Add the release note for 1.6.0

### PR DESCRIPTION
**Pull request by minwoox**

### CODE REVIEW

**Comment by minwoox** [comment]
screencapture-localhost-8000-release-notes-1-6-0-2021-04-06-14_50_30

File: site/src/pages/release-notes/1.6.0.mdx [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+- You can now fluently build a <type://DecodingClient> using the <type://DecodingClientBuilder>. #3348 #3372
+  ```java
+  DecodingClient.builder()
+                .autoFillAcceptEncoding(false)
+                .strictContentEncoding(true)
+                .newDecorator();
+  ```
+- You can now fluently build an <type://HttpRequest> with a `Publisher<HttpData>`. #3343
+  ```java
+  Publisher<HttpData> publisher = StreamMessage.of(...);
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:113]: Just?
```suggestion
  StreamMessage<HttpData> publisher = StreamMessage.of(...);
```

File: site/src/pages/release-notes/1.6.0.mdx [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+  ```java
+  new MyFilteredHttpResponse(res) {
+      ...
+      @Override
+      protected void onCancellation(Subscriber<? super U> subscriber) {
+          // Clean up resources.
+      }
+  }
+  ```
+- You can now easily set 'cookie' or 'set-cookie' headers. #3391
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:180]: The getters are also added.

File: site/src/pages/release-notes/1.6.0.mdx [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+                                         .cookie(cookie)
+                                         .build();
+
+  Cookie setCookie1 = ...
+  Cookie setCookie2 = ...
+  ResponseHeaders headers = ResponseHeaders.builder(HttpStatus.OK)
+                                           .cookies(setCookie1, setCookie2)
+                                           .build();
+  ```
+- You can now use `req.root_id` <type://BuiltInProperty> to log the id of <type://RequestContext#root()>.
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:193]:
```suggestion
- You can now use `req.root_id` <type://BuiltInProperty> to log the ID of <type://RequestContext#root()>.
```

File: site/src/pages/release-notes/1.6.0.mdx [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+  #3429 #3433
+- You can now use Thrift 0.14.0. #3470 #3422
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:195]: ?
```suggestion
- You can now use Thrift 0.14.1. #3470 #3422
```

File: site/src/pages/release-notes/1.6.0.mdx [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+- Various improvements for documentation. #3237 #3289 #3361 #3390 #3401 #3427
+
+## 🛠️ Bug fixes
+
+- You no longer see `NullPointerException` in <type://HttpResponseDecoder>. #3036 #3407
+- You no longer see 405 when the exact and param path are defined with different HTTP methods. #3330 #3340
+- You no longer see `Address family not supported by protocol` or `Connection refused` error anymore. #3425
+- You no longer see escaped Unicode from a JSON response in <type://DocService>. #3396
+- You no longer see the wrong response body when the payload violates the protocol or is too large. #3419
+- You can now use the Thrift client and service that is generated by `java:fullcamel` Thrift generator.
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:231]:
```suggestion
- You can now use the Thrift client and service that is generated by [`java:fullcamel`](https://issues.apache.org/jira/browse/THRIFT-2469) option of Thrift compiler.
```

File: site/src/pages/release-notes/1.6.0.mdx [hunk 6]
```diff
@@ -0,0 +1,315 @@
+---
+date: 2021-04-06
+---
+
+## 🌟 New features
+
+- You can now send and handle the multipart request. #253 #3327
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:7]: multipart requests and responses? ('the' is not necessary)

File: site/src/pages/release-notes/1.6.0.mdx [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+
+      if (cause instanceof IllegalArgumentException) {
+          return AggregatedHttpResponse.of(HttpStatus.BAD_REQUEST);
+      }
+
+      // Return null to let ExceptionHandler.ofDefault() convert the exception.
+      return null;
+  }
+
+  Server.builder().exceptionHandler(handler)...
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:61]: How about simplifying like this:

```
Server.builder().exceptionHandler((ctx, cause) -> {
    ...
})...
```

File: site/src/pages/release-notes/1.6.0.mdx [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+  ```
+- You can now convert an exception into an <type://RpcResponse> in <type://THttpService>. #3349 #3383
+  ```java
+  THttpService.builder()
+              .addService("hello", helloService)
+              .exceptionHandler((ctx, cause) -> {
+                  if (cause instanceof IllegalArgumentException) {
+                      return RpcResponse.of(CustomizedException("Bad Request!"));
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:69]: `new CustomizedException(...)`?

File: site/src/pages/release-notes/1.6.0.mdx [hunk 9]
```diff
@@ -0,0 +1,315 @@
...
+  ```java
+  WebClient client = ...;
+  client.prepare()
+        .get("/my-service")
+        .responseTimeout(Duration.ofSeconds(3))
+        .attr(USER_ID, userId)
+        .attr(USER_SECRET, secret)
+        .execute();
+  ```
+- You can now add additional error details to the response when an exception is raised in gRPC service.
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:84]: - add additional -> specify additional ? (not two 'add's)
- the response -> a response (or a gRPC response)
- a gRPC service

File: site/src/pages/release-notes/1.6.0.mdx [hunk 10]
```diff
@@ -0,0 +1,315 @@
...
+  ```java
+  ServerBuilder sb = ...
+  sb.decorator((delegate, ctx, req) -> {
+      if (ctx.config().route().isFallback()) {
+          // This request is not matched any routes.
+      }
+      return delegate.serve(ctx, req);
+  });
+  ```
+- The <type://BraveService> does not trace requests to the <type://TransientService> such as
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:160]: - The BraveService -> BraveService
- to the -> for

File: site/src/pages/release-notes/1.6.0.mdx (hunk 4 above)
Comment (trustin) [reply site/src/pages/release-notes/1.6.0.mdx:195]: How about:

    You can now use Thrift 0.14.0 with new `armeria-thrift0.14` module.

File: site/src/pages/release-notes/1.6.0.mdx [hunk 11]
```diff
@@ -0,0 +1,315 @@
...
+  Cookie setCookie1 = ...
+  Cookie setCookie2 = ...
+  ResponseHeaders headers = ResponseHeaders.builder(HttpStatus.OK)
+                                           .cookies(setCookie1, setCookie2)
+                                           .build();
+  ```
+- You can now use `req.root_id` <type://BuiltInProperty> to log the id of <type://RequestContext#root()>.
+  #3429 #3433
+- You can now use Thrift 0.14.0. #3470 #3422
+- You can now use OAuth 2.0 related features. #2268 #2840
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:196]: .. with new `armeria-oauth2` module.

File: site/src/pages/release-notes/1.6.0.mdx [hunk 12]
```diff
@@ -0,0 +1,315 @@
...
+  </Tip>
+
+## 📈 Improvements
+
+- Various improvements for documentation. #3237 #3289 #3361 #3390 #3401 #3427
+
+## 🛠️ Bug fixes
+
+- You no longer see `NullPointerException` in <type://HttpResponseDecoder>. #3036 #3407
+- You no longer see 405 when the exact and param path are defined with different HTTP methods. #3330 #3340
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:227]: 405 -> `405 Method Not Allowed` status

File: site/src/pages/release-notes/1.6.0.mdx [hunk 13]
```diff
@@ -0,0 +1,315 @@
...
+- You no longer see `Address family not supported by protocol` or `Connection refused` error anymore. #3425
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:228]: anymore on certain machines with IPv6 enabled.

File: site/src/pages/release-notes/1.6.0.mdx [hunk 14]
```diff
@@ -0,0 +1,315 @@
...
+- You no longer see escaped Unicode from a JSON response in <type://DocService>. #3396
+- You no longer see the wrong response body when the payload violates the protocol or is too large. #3419
+- You can now use the Thrift client and service that is generated by `java:fullcamel` Thrift generator.
+  #3269 #3360 #3369
+- `SmartLifecycle` for Armeria server graceful shutdown is only created when the Armeria server is created
+  by Spring integration. #3300
+- You can now use [Sealed oneofs](https://scalapb.github.io/docs/sealed-oneofs/) message from ScalaPB
+  using JSON. #3342
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:236]:
```suggestion
- You can now use [Sealed oneofs](https://scalapb.github.io/docs/sealed-oneofs/) message from ScalaPB
  with JSON for gRPC and annotated services. #3342 #3394
```
Add #3394 too?

File: site/src/pages/release-notes/1.6.0.mdx [hunk 15]
```diff
@@ -0,0 +1,315 @@
...
+## 📈 Improvements
+
+- Various improvements for documentation. #3237 #3289 #3361 #3390 #3401 #3427
+
+## 🛠️ Bug fixes
+
+- You no longer see `NullPointerException` in <type://HttpResponseDecoder>. #3036 #3407
+- You no longer see 405 when the exact and param path are defined with different HTTP methods. #3330 #3340
+- You no longer see `Address family not supported by protocol` or `Connection refused` error anymore. #3425
+- You no longer see escaped Unicode from a JSON response in <type://DocService>. #3396
```
Comment (trustin) [inline site/src/pages/release-notes/1.6.0.mdx:229]: The Unicode characters like emojis in a JSON response are now rendered correctly in DocService. ?

**Comment by minwoox** [comment]
@trustin and @ikhoon Thanks for the review. All fixed. PTAL. 😄

**Review by ikhoon — APPROVED** [review]
Huge additions! Thanks, @minwoox. 🙇‍♂️

**Review by trustin — APPROVED** [review]
Thanks a lot, @minwoox ! Excited about this release :tada:

File: site/src/pages/release-notes/1.6.0.mdx [hunk 16]
```diff
@@ -0,0 +1,315 @@
...
+    'eugene70',
+    'ghkim3221',
+    'hyangtack',
+    'heka1024',
+    'heowc',
+    'hexoul',
+    'ikhoon',
+    'JunoJunho',
+    'kojilin',
+    'ks-kim',
```
Comment (ikhoon) [inline site/src/pages/release-notes/1.6.0.mdx:302]: Oops... His GitHub ID was https://github.com/ks-yim . 😱

**Pull request merged**